pkg/agent/util/net_windows.go

```diff
@@ -704,20 +656,56 @@ func SetInterfaceMTU(ifaceName string, mtu int) error {
 }
 
 func NewNetRoute(route *Route) error {
-	return route.Add()
+	row := route.toMibIPForwardRow()
+	if err := createIPForwardEntry(row); err != nil {
+		return fmt.Errorf("failed to create new IPForward row: %v", err)
+	}
+	return nil
 }
 
 func RemoveNetRoute(route *Route) error {
-	return route.Delete()
+	family := addressFamilyByIP(route.DestinationSubnet.IP)
+	rows, err := listIPForwardRows(family)
+	if err != nil {
+		return fmt.Errorf("unable to list Windows IPForward rows: %v", err)
+	}
+	for i := range rows {
+		row := rows[i]
+		if row.DestinationPrefix.EqualsTo(route.DestinationSubnet) && row.Index == uint32(route.LinkIndex) && row.NextHop.IP().Equal(route.GatewayAddress) {
```

Comment:
Could you help verify if this comparasion works for Windows default route ( in which GatewayAddress may be empty)?

Reply:
Note that the previous command doesn't check if gatewyAddress matches, so we should confirm if this still works as expected.

Reply:
Verified on Windows host, the existing logic can be used on default routes and the Windows auto-generated routes which uses 0.0.0.0 ("on-link" shown in command route print) as Gateway